ledger/participant-integration-api/src/test/lib/scala/platform/store/dao/JdbcLedgerDaoEventsSpec.scala

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao
+
+import com.daml.lf.ledger.EventId
+import com.daml.lf.transaction.{GlobalKey, GlobalKeyWithMaintainers}
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.{Inside, LoneElement, OptionValues}
+
+import scala.concurrent.Future
+
+private[dao] trait JdbcLedgerDaoEventsSpec extends LoneElement with Inside with OptionValues {
+  this: AsyncFlatSpec with Matchers with JdbcLedgerDaoSuite =>
+
+  private def eventsReader = ledgerDao.eventsReader
+
+  private def globalKeyWithMaintainers(value: String) = GlobalKeyWithMaintainers(
+    GlobalKey.assertBuild(someTemplateId, someContractKey(alice, value)),
+    Set(alice),
+  )
+
+  behavior of "JdbcLedgerDao (events)"
+
+  it should "lookup a create event by contract id" in {
+
+    for {
+      (_, tx) <- store(singleCreate(cId => create(cId)))
+      flatTx <- ledgerDao.transactionsReader.lookupFlatTransactionById(
+        tx.transactionId,
+        tx.actAs.toSet,
+      )
+      result <- eventsReader.getEventsByContractId(
+        nonTransient(tx).loneElement,
+        Set(alice),
+      )
+    } yield {
+      val expected = flatTx.value.transaction.value.events.loneElement.event.created.value
+      val actual = result.createEvent.value
+      actual shouldBe expected
+    }
+  }
+
+  it should "lookup an archive event by contract id" in {
+    for {
+      (_, tx1) <- store(singleCreate(cId => create(cId)))
+      contractId = nonTransient(tx1).loneElement
+      (_, tx2) <- store(singleExercise(contractId))
+      flatTx <- ledgerDao.transactionsReader.lookupFlatTransactionById(
+        tx2.transactionId,
+        tx2.actAs.toSet,
+      )
+      expected = flatTx.value.transaction.value.events.loneElement.event.archived.value
+      result <- eventsReader.getEventsByContractId(contractId, Set(alice))
+    } yield {
+      val actual = result.archiveEvent.value
+      actual shouldBe expected
+    }
+  }
+
+  it should "make events visible to signatories and observers (stakeholders) only" in {
+
+    for {
+      (_, tx) <- store(
+        singleCreate(cId => create(cId, signatories = Set(alice), observers = Set(charlie)))
```

Comment:
I think here we are proving that a party uninvolved in a transaction (Emma) doesn't see the events. We are not proving that a party involved in a transaction (an actor perhaps) is not able to see the events. I think the `singleCreate` method can be used to also provide a different actor.